### ISSUE TITLE
Auto limit of 1000 breaks Oracle queries

### ISSUE THREAD

**Issue by jhult** [getredash/redash#5180]
### Issue Summary

Pull request #5088 introduced the ability to auto limit SQL results to the first 1000 results (#976). Using this feature breaks Oracle queries. This is because Oracle does not support the `LIMIT` clause. Instead, Oracle uses another mechanism (see [here](https://oracle-base.com/articles/misc/top-n-queries) for details).

### Steps to Reproduce

1. Setup an instance with the Oracle driver
2. Run any query and check the "LIMIT 1000" box
3. Receive this error: `Error running query: Unknown error occurred. Please try again later.`

![image](https://user-images.githubusercontent.com/9849069/93403849-a063ea80-f856-11ea-8b4f-133bb65f5cf4.png)

### What did you expect to happen instead?

This feature should be adjusted to work on Oracle queries.

### Technical details:

* Redash Version: master branch (plus [this commit](https://github.com/getredash/redash/pull/5179) to get around separate Oracle bug)
* Browser/OS: Brave / macOS
* How did you install Redash: [Helm chart](https://github.com/getredash/contrib-helm-chart)

**Comment by tim5go** [comment]
@jhult 
Oracle supports LIMIT with a different syntax:
`FETCH NEXT 1000 ROWS ONLY;`

**Comment by jhult** [comment]
@tim5go, yes, Oracle supports limiting the number of results. But, as you mentioned, not using the exact same syntax (which is why this currently breaks Oracle queries).

One option is what you mentioned. Some other possibilities can be found in the link in my original post.

**Comment by tim5go** [comment]
@jhult 
I opened a PR #5181 for that

**Comment by vshivkova** [comment]
Auto limit of 1000 breaks MSSQL queries too. MSSQL does not support LIMIT syntax. Is uses TOP or OFFSET or SET ROWCOUNT.

**Comment by jonahgreenthal** [comment]
Any hope of getting this fixed soon? Having to uncheck that box for every query is a really annoying extra step.
